src/main/java/com/appdynamics/monitors/azure/AzureMonitor.java

```diff
@@ -8,139 +8,43 @@
-import com.fasterxml.jackson.databind.JsonNode;
-import com.fasterxml.jackson.databind.node.ArrayNode;
-import com.singularity.ee.agent.systemagent.api.AManagedMonitor;
-import com.singularity.ee.agent.systemagent.api.TaskExecutionContext;
-import com.singularity.ee.agent.systemagent.api.TaskOutput;
-import com.singularity.ee.agent.systemagent.api.exception.TaskExecutionException;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import java.net.URL;
-import java.util.ArrayList;
+import com.appdynamics.extensions.ABaseMonitor;
+import com.appdynamics.extensions.TasksExecutionServiceProvider;
+import com.appdynamics.extensions.util.AssertUtils;
+import com.appdynamics.monitors.azure.utils.Constants;
+import java.util.List;
 import java.util.Map;
+import java.util.concurrent.CountDownLatch;
 
-@SuppressWarnings("WeakerAccess")
-public class AzureMonitor extends AManagedMonitor {
-    private static final Logger logger = LoggerFactory.getLogger(AzureMonitor.class);
-    private MonitorConfiguration configuration;
+@SuppressWarnings({"WeakerAccess", "unchecked"})
+public class AzureMonitor extends ABaseMonitor {
 
-    public AzureMonitor() { logger.info(String.format("Using Azure Monitor Version [%s]", getImplementationVersion())); }
-
-    private void initialize(Map<String, String> argsMap) {
-        if (configuration == null) {
-            MetricWriteHelper metricWriteHelper = MetricWriteHelperFactory.create(this);
-            MonitorConfiguration conf = new MonitorConfiguration(Globals.defaultMetricPrefix,
-                    new TaskRunnable(), metricWriteHelper);
-            final String configFilePath = argsMap.get(Globals.configFile);
-            conf.setConfigYml(configFilePath);
-            conf.setMetricWriter(MetricWriteHelperFactory.create(this));
-            conf.checkIfInitialized(ConfItem.CONFIG_YML,
-                    ConfItem.EXECUTOR_SERVICE,
-                    ConfItem.METRIC_PREFIX,
-                    ConfItem.METRIC_WRITE_HELPER);
-            this.configuration = conf;
-        }
+    @Override
+    protected String getDefaultMetricPrefix() {
+        return Constants.DEFAULT_METRIC_PREFIX;
     }
 
-    public TaskOutput execute(Map<String, String> map, TaskExecutionContext taskExecutionContext) throws TaskExecutionException {
-        try{
-            if(map != null){
-                if (logger.isDebugEnabled()) {logger.debug("The raw arguments are {}", map);}
-                initialize(map);
-                configuration.executeTask();
-                return new TaskOutput("Azure Monitor Metric Upload Complete");
-            }
-        }
-        catch(Exception e) {
-            logger.error("Failed to execute the Azure monitoring task", e);
-        }
-        throw new TaskExecutionException("Azure monitoring task completed with failures.");
-
+    @Override
+    public String getMonitorName() {
+        return "Azure Monitoring Extension";
```

Comment:
"Azure Monitoring Extension" can also go to the Constants class, just like the Default Metric Prefix above.